src/Socket.php

```diff
@@ -36,16 +36,12 @@ public function __construct(LoopInterface $loop, $socket, Buffer $buffer = null)
 
     public function getLocalAddress()
     {
-        if ($this->socket !== false) {
-            return stream_socket_get_name($this->socket, false);
-        }
+        return $this->sanitizeAddress(@stream_socket_get_name($this->socket, false));
```

Comment:
No worries, I actually appreciate these kind of discussions :-)

This PR makes it so that we *ignore* any warnings (which are unreliable) and simply rely on the function returning `false` (which is documented behavior and works reliably across different PHP versions and HHVM). This is covered by both our test suite and the above gist.

Once this PR is in, the behavior is consistent throughout the Datagram component.

It is currently not consistent with regards to the Socket component. As you rightfully pointed out, the Socket component does currently not implement any checks and does not suppress any errors either. This means that this does in fact error out when one tries to access to remote address after the socket has been closed. See also https://github.com/reactphp/socket/issues/19, for which I've prepared a PR that I'll file once https://github.com/reactphp/socket/pull/61 is in :+1: